dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -178,6 +150,62 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
         }
     }
 
+    void AppNotificationManager::RegisterActivationCallback(winrt::guid const& registeredClsid)
+    {
+        auto lock{ m_lock.lock_exclusive() };
+        THROW_HR_IF_MSG(E_INVALIDARG, m_notificationComActivatorRegistration, "Already Registered for App Notifications!");
+
+        // Check if the caller has registered event handlers, if so the REGCLS_MULTIPLEUSE flag will cause COM to ensure that all activators
+        // are routed inproc, otherwise with REGCLS_SINGLEUSE COM will launch a new process of the Win32 app for each invocation.
+        auto activationFlag{ m_notificationHandlers ? REGCLS_MULTIPLEUSE : REGCLS_SINGLEUSE };
+
+        // Register an INotificationActivationCallback to receive background activations from AppNotification.
+        // Also, STA threads that call CoRegisterClassObject need to use the REGCLS_AGILE flag so that the object is
+        // associated with the neutral apartment. This allows other threads to activate the STA registered thread.
+        THROW_IF_FAILED(::CoRegisterClassObject(
+            registeredClsid,
+            winrt::make<AppNotificationManagerFactory>().get(),
+            CLSCTX_LOCAL_SERVER,
+            activationFlag | REGCLS_AGILE,
+            &m_notificationComActivatorRegistration));
+    }
+
+    void AppNotificationManager::RegisterAppNotificationSinkWithNotificationPlatform()
+    {
+        auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+        THROW_IF_FAILED(notificationPlatform->AddToastRegistrationMapping(m_processName.c_str(), m_appId.c_str()));
+    }
+
+    winrt::guid AppNotificationManager::RegisterPackagedApp()
+    {
+        winrt::guid registeredClsid{ PushNotificationHelpers::GetComRegistrationFromRegistry(c_expectedAppServerArgs.data()) };
+
+        if (!PushNotificationHelpers::IsPackagedAppScenario() && !WindowsAppRuntime::SelfContained::IsSelfContained())
+        {
+            RegisterAppNotificationSinkWithNotificationPlatform();
+        }
+
+        return registeredClsid;
+    }
+
+    winrt::guid AppNotificationManager::RegisterUnpackagedApp(AppNotificationAssets const& assets)
+    {
+        THROW_IF_FAILED(PushNotifications_RegisterFullTrustApplication(m_appId.c_str(), GUID_NULL));
+
+        std::wstring comActivatorGuidString{ GetOrCreateComActivatorGuid() };
+        std::wstring notificationAppId{ RetrieveNotificationAppId() };
+
+        RegisterAssets(notificationAppId, comActivatorGuidString, assets);
+
+        if (!WindowsAppRuntime::SelfContained::IsSelfContained())
+        {
+            RegisterAppNotificationSinkWithNotificationPlatform();
+        }
+
+        // Remove braces around the guid string
+        return winrt::guid(comActivatorGuidString.substr(1, comActivatorGuidString.size() - 2));
```

Comment:
Would be great if the helper above took care of braces removal. Is that possible?